andela/wg-rogue-one
# #157731614 Add Dropset Support

### What does this PR do?
This PR creates a feature that enables users to add dropset workouts.

Add `changeWorkoutTyp`e function to `wger/core/static/js/wger-core.js`
Add a onClick event handler to function to `addEntry button` in `wger/manager/templates/day/log.html`
Add functionality to enable adding new sets to the existing sets in `wger/manager/views/log.py`
Create new model WorkoutType, add category field to Setting model as a
foreign key.

#### Description of Task to be completed?
Add support for dropsets((https://en.wikipedia.org/wiki/Drop_set)

#### How should this be manually tested?
1: Clone the application and install the requirements in your virtual environment.
2: Update the database settings in the `settings.py` file with your postgres user, password and database name.
3: Run the application with `python manage.py runserver`.
4: Register and login.
5: Hover over `Workouts` in the navbar menu and select `Workouts`.
6: Add a `workout` then a `day` and `description` then add an `exercise` to the  workout: 
7: Add an initial weight and amount
8: You should be able to declare a workout type as a `dropset` or `default` workout.

##### Expected response
<img width="1218" alt="screen shot 2018-06-17 at 16 37 30" src="https://user-images.githubusercontent.com/25340897/41508407-236110f8-724d-11e8-96eb-4c7bfb1a6cea.png">

8: Click save and you should view the created workout as a dropset workout.

##### Expected response
<img width="1270" alt="screen shot 2018-06-17 at 16 37 57" src="https://user-images.githubusercontent.com/25340897/41508408-275c413c-724d-11e8-9357-a3e65a0f6353.png">

9: Go to `Workouts`-> `carlendar` and add new log to the workout of the day respectively.
10: You should be able to add a new dropset reports while you work out.

##### Expected response
<img width="1228" alt="screen shot 2018-06-17 at 16 38 41" src="https://user-images.githubusercontent.com/25340897/41508411-2b6bd4fe-724d-11e8-80ce-ef781cb4e674.png">

#### Any background context you want to provide?
This feature was not available.

#### What are the relevant pivotal tracker stories?
#157731614
